### PR TITLE
yaz and friends: update

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

@ryandesign Looks like I cannot re-open the old PR, so I just make a new one.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by barracuda156** [comment]
Won’t build yet, need to rebase a patch and fix this:
```
  pazpar2.c:79:5: error: call to undeclared library function 'exit' with type 'void (int) __attribute__((noreturn))'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
      exit(0);
      ^
  pazpar2.c:79:5: note: include the header <stdlib.h> or explicitly provide a declaration for 'exit'
  1 error generated.
  make[2]: *** [pazpar2.o] Error 1
```

File: net/yazproxy/files/dynamic_lookup-11.patch [hunk 1]
```diff
@@ -1,26 +1,31 @@
...
-+++ configure	2024-05-23 14:10:02.000000000 -0500
-@@ -7923,16 +7923,11 @@
-       _lt_dar_allow_undefined='${wl}-undefined ${wl}suppress' ;;
+--- configure	2022-08-18 16:32:05
++++ configure	2024-06-06 12:59:04
+@@ -9083,18 +9083,13 @@
+       _lt_dar_allow_undefined='$wl-undefined ${wl}suppress' ;;
      darwin1.*)
-       _lt_dar_allow_undefined='${wl}-flat_namespace ${wl}-undefined ${wl}suppress' ;;
+       _lt_dar_allow_undefined='$wl-flat_namespace $wl-undefined ${wl}suppress' ;;
```
Comment (ryandesign) [inline net/yazproxy/files/dynamic_lookup-11.patch:9]: There's no need to change this line. The curly braces around `{wl}` are not necessary in these two instances but are not harmful either. There have been many changes in libtool over the years and thus many subtle variations in the dynamic_lookup-11.patch files in MacPorts. Our goal is not to completely change a project's code to match the latest libtool, but adapt the patch so that it makes the minimal changes to accomplish the goal.

File: net/yazproxy/files/dynamic_lookup-11.patch [hunk 2]
```diff
@@ -1,26 +1,31 @@
...
+-      esac
 +    darwin*)
 +      case ${MACOSX_DEPLOYMENT_TARGET},$host in
-+	10.[012],*|,*powerpc*)
- 	  _lt_dar_allow_undefined='${wl}-flat_namespace ${wl}-undefined ${wl}suppress' ;;
--	10.*)
-+	*)
- 	  _lt_dar_allow_undefined='${wl}-undefined ${wl}dynamic_lookup' ;;
-       esac
++    10.[012],*|,*powerpc*-darwin8*)
```
Comment (ryandesign) [inline net/yazproxy/files/dynamic_lookup-11.patch:24]: It makes no difference for MacPorts since it always sets `MACOSX_DEPLOYMENT_TARGET` and since MacPorts doesn't support Darwin < 8 but for consistency with the code produced by libtool 2.4.7 and later this should read:
```suggestion
+    10.[012],*|,*powerpc*-darwin[5-8]*)
```
See https://debbugs.gnu.org/cgi/bugreport.cgi?bug=44605#20.

File: net/yazproxy/files/dynamic_lookup-11.patch [hunk 3]
```diff
@@ -1,26 +1,31 @@
...
++      _lt_dar_allow_undefined='${wl}-flat_namespace ${wl}-undefined ${wl}suppress' ;;
++    *)
++      _lt_dar_allow_undefined='${wl}-undefined ${wl}dynamic_lookup' ;;
++      esac
```
Comment (ryandesign) [inline net/yazproxy/files/dynamic_lookup-11.patch:28]: There's no need to change these lines just for whitespace reasons. Let's try to keep the number of variations of dynamic_lookup-11.patch to a minumum by not adding unnecessary changes.

**Review by ryandesign — COMMENTED** [review]
Did the patches you added for pazpar2 and zebra come from upstream? If not, can you submit them there?

**Comment by pmetzger** [comment]
@barracuda156 Please address @ryandesign's comments?

**Comment by barracuda156** [comment]
@pmetzger Yeah, I did not forget, just too much stuff to do. Will deal with this one shortly.

**Comment by pmetzger** [comment]
Poke.

**Comment by barracuda156** [comment]
@pmetzger Will finalize this no later than this weekend. Sorry, got stuck today with other stuff.
(I won’t drop it, we spent too much time on this.)

**Comment by barracuda156** [comment]
(Ok, I am finally back to this one.)

**Comment by pmetzger** [comment]
CI failed.

**Comment by barracuda156** [comment]
Yeah, I will look into that.

**Comment by barracuda156** [comment]
UPD. Ah, it is `libxml2` breaking things again :( 
It was updated: https://github.com/macports/macports-ports/commit/62ef22bf7838a9ac4572b1ea85aa051623d26129

@ryandesign Do we know what else may be broken? _A lot_ of ports use it.

**Comment by pmetzger** [comment]
Poke.

**Comment by barracuda156** [comment]
@pmetzger Well, we have a breakage due to `libxml2` update now -_-
We need to find a fix – this issue was not present originally.

**Comment by pmetzger** [comment]
So what is the next step?

**Comment by barracuda156** [comment]
> So what is the next step?

I need to find time for that (provided nobody else is interested). It is not under my control what `libxml2` upstream decides to break in their API all of a sudden.

**Comment by ryandesign** [comment]
metaproxy failed to build:

```
/opt/local/include/libxml2/libxml/xinclude.h:120:7: error: unknown type name 'xmlStructuredErrorFunc'
                                         xmlStructuredErrorFunc handler,
                                         ^
```

Probably needs `#include <libxml/xmlerror.h>`.

-----

pazpar2 failed to build:

```
pazpar2.c:113:5: error: call to undeclared function 'xmlInitParser'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
    xmlInitParser();
    ^
1 error generated.
```

Probably needs `#include <libxml/parser.h>`.

-----

zebra failed to build:

```
zebramap.c:243:19: error: call to undeclared function 'xmlParseFile'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        zm->doc = xmlParseFile(full_path);
                  ^
```

Probably needs `#include <libxml/parser.h>`.

-----

These are upstream issues that need to be reported to these projects. libxml2 2.12.0 cleaned up its headers to remove unnecessary inclusions, which exposes areas where projects that use libxml2 haven't included all the headers they need to.

**Comment by ryandesign** [comment]
> metaproxy failed to build:
> 
> ```
> /opt/local/include/libxml2/libxml/xinclude.h:120:7: error: unknown type name 'xmlStructuredErrorFunc'
>                                          xmlStructuredErrorFunc handler,
>                                          ^
> ```
> 
> Probably needs `#include <libxml/xmlerror.h>`.

This is fixed in libxml2 2.13.1.

https://gitlab.gnome.org/GNOME/libxml2/-/commit/95939d6ea3718c458620eeda850add549cd07e99

File: databases/pazpar2/files/patch-includes.diff
Comment (ryandesign) [inline databases/pazpar2/files/patch-includes.diff:1]: I've submitted a PR for this here: https://github.com/indexdata/pazpar2/pull/36

File: databases/zebra/Portfile [hunk 4]
```diff
@@ -16,17 +17,19 @@ long_description    Zebra is a high-performance, general-purpose \
                     through exact boolean search expressions \
                     and relevance-ranked free-text queries.
 
-homepage            https://www.indexdata.com/resources/software/zebra/
+homepage            https://www.indexdata.com/resources/software/zebra
```
Comment (ryandesign) [inline databases/zebra/Portfile:20]: The canonical URL at this server does have the trailing slash so this change should be reverted.
```suggestion
homepage            https://www.indexdata.com/resources/software/zebra/
```

**Comment by ryandesign** [comment]
> This is fixed in libxml2 2.13.1 (#24676)

Now that this is merged you can rebase from master to fix this one but will still need patches for the other two.

**Comment by pmetzger** [comment]
Are we ready to merge now?

**Comment by barracuda156** [comment]
I hope so. Looks like libxml2 issues are resolved.
